### PR TITLE
validator: add role and helpers to install

### PR DESCRIPTION
**Pull request by fromanirh**
Add support to deploy the template-validator

Signed-off-by: Francesco Romani <fromani@redhat.com>

### CODE REVIEW

**Comment by fromanirh** [comment]
tested on kubevirt (not okd/ocp). The webhook is succesfully deployed. Then crashes, because there is not the template resource. But the deploy process itself works.

File: roles/KubevirtTemplateValidator/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  k8s:
+    state: present
+    src: "{{ certsdata.webhookcsr }}"
+# TODO: wait for CSR to be created
+- name: Approve CSR
+  command: /usr/bin/oc adm certificate approve {{ certsdata.csrname }}
+- name: Fetch certificate from CSR
+  set_fact:
+    # TODO: certsdata.csrname doesn't work. Maybe different parameter handling?
+    csrdata: "{{ lookup('k8s', kind='CertificateSigningRequest', api_version='certificates.k8s.io/v1beta1', resource_name='virtualmachine-template-validator.kubevirt') }}"
```
Comment (MarSik) [inline roles/KubevirtTemplateValidator/tasks/main.yml]: You can also use k8s_facts module and register: the output, lookup works too though

File: roles/KubevirtTemplateValidator/tests/test.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
```
Comment (MarSik) [inline roles/KubevirtTemplateValidator/tests/test.yml]: I do not think we need this file, it won't work without the deploy manifests anyway

Comment (fromanirh) [reply roles/KubevirtTemplateValidator/tests/test.yml]: Removed

File: roles/KubevirtTemplateValidator/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
+---
+# tasks file for KubevirtTemplateValidator
+- name: Set template:view role
+  k8s:
+    state: present
+    definition: "{{ lookup('file', 'template-view-role.yaml') }}"
```
[the remark this answers is not in the document]
Comment (fromanirh) [reply roles/KubevirtTemplateValidator/tasks/main.yml]: AFAIK clusterroles aren't namespaced

File: roles/KubevirtTemplateValidator/tasks/main.yml (hunk 1 above)
Comment (fromanirh) [reply roles/KubevirtTemplateValidator/tasks/main.yml]: Seems much nicer, thanks! done

**Comment by fromanirh** [comment]
obsoleted by https://github.com/MarSik/kubevirt-ssp-operator/pull/6

**Pull request closed without merging**